### PR TITLE
[WIP] Feature termcaps

### PR DESCRIPTION
**Pull request by jgigault**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,6 +20,8 @@ NAME		=	42sh
 
 SRCS		=	\
 				main.c							\
+				builtins/cd/builtin_cd.c		\
+				builtins/builtin_callback.c		\
 				exit/shell_exit.c				\
```
Comment (jgigault) [inline Makefile:25]: it should be removed too

File: srcs/exit/shell_exit.c [hunk 2]
```diff
@@ -6,9 +6,12 @@
 ** - (TODO) set the termios structure to its original value
```
Comment (jgigault) [inline srcs/exit/shell_exit.c:6]: File has to be renamed :-)

File: incs/shell.h [hunk 3]
```diff
@@ -23,19 +36,25 @@
...
+** builtins/
+*/
+int				builtin_callback(int callback, t_sh *sh, t_proc *p);
 
 /*
-** exit/
+** log_st /
 */
-void			shell_exit(int status);
+void			log_st(int status);
```
Comment (jgigault) [inline incs/shell.h:57]: what about `log_status()` instead?
and could you add an additional `char *suffix` as talked about today?

File: srcs/signal/signal_sigchld.c [hunk 4]
```diff
@@ -52,18 +52,16 @@ void		signal_sigchld(int sig)
...
 	{
 		log_info("SIGCHLD received");
 		errno = 0;
-		while (test == 0)
+		while (1)
 		{
 			pid = waitpid(WAIT_ANY, &status, WUNTRACED | WNOHANG);
 			if (proc_update_status(pid, status) == 0)
-				test = 1;
+				break ;
```
Comment (gabkk) [inline srcs/signal/signal_sigchld.c:64]: :+1:

File: srcs/termcaps/termcaps_read_input.c [hunk 5]
```diff
@@ -58,76 +58,83 @@ static int	s_termcaps_identify_input(const unsigned int c,
...
 	{
 		ASSERT(termcaps_string_to_command_line(PROMPT_SIZE,
 												PROMPT, &context.command_line));
 		ASSERT(termcaps_display_command_line(&context.command_line));
 	}
-	else if (context.state != STATE_REGULAR)
+	else if (context.state == STATE_CONTINUE)
 	{
-		write (shell_fd(), "\r", 2);
+		caps__print_cap(CAPS__CARRIAGE_RETURN, 0);
```
Comment (gabkk) [inline srcs/termcaps/termcaps_read_input.c:95]: :100:

File: Makefile (hunk 1 above)
Comment (abombard) [inline Makefile:25]: yes je m'occupe de tout ca je voulais juste voir si ca marchait vite fait

**Pull request closed without merging**